packages/core/content-manager/server/src/services/uid.ts

```diff
@@ -73,15 +82,17 @@ export default ({ strapi }: { strapi: Strapi }) => ({
     contentTypeUID,
     field,
     value,
+    locale,
   }: {
     contentTypeUID: UID.ContentType;
     field: string;
     value: string;
+    locale: string;
   }) {
     const query = strapi.db.query(contentTypeUID);
 
     const count: number = await query.count({
-      where: { [field]: value },
+      where: { [field]: value, locale },
     });
```

Comment:
shouldn't we consider the publication state too? a UID could be the same in a draft and publish version.

Reply:
I'm not sure. If you're on a published entry and you change the UID you could then also save

I think for these generating and checking endpoints we should either check across all entries (when in a published state) or just drafts otherwise, wdyt?

Reply:
Mmmm thats a good question, my intuition is saying it could behave the same as the uniqueness feature, things are unique accross the same publication state and the same locale.

In any case, this endpoint is only triggered for draft data right? Published data is not editable directly if not publishing a draft.

Reply:
Okay, so do you think we should only check across draft entries?

The only confusion then could be that the field says it's available but the validation fails when you go to publish

Reply:
Thats a very good point, and that would also involve unique fields. 
Thats maybe a UX question we could raise to the team. Maybe you are right, and  it does make sense for unique fields to only be unique accross locales? 🤔

Reply:
Okay, for now I think I will build this to only check across draft entries. Then we have fairly consistent behaviour everywhere and I will track the UX question elsewhere, wdyt?

Reply:
sounds good!!